### PR TITLE
VLAN tests wo traffic

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Tests:
 - test_max_vlans;
 - test_add_interface_to_vlan_wo_bridge
Signed-off-by: Stepan Vovk <stepan.vovk@plvision.eu>

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import json
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: We should not group tests based on the traffic-sending criteria. Tests should be grouped based on the feature.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+from dent_os_testbed.lib.bridge.bridge_vlan import BridgeVlan
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.utils.test_utils.tgen_utils import tgen_utils_get_dent_devices_with_tgen
+
+pytestmark = pytest.mark.suite_functional_vlan
+
+
+@pytest.mark.asyncio
+async def test_max_vlans(testbed):
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: E.g. This one belongs to VLAN scaling.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    """
+    Test Name: Maximum vlans for the interface
+    Test Suite: suite_functional_vlan
+    Test Overview: Test maximum number of vlans that can be set on interface
+    Test Procedure:
+    1. Initiate test params
+    2. Create bridge entity and set state to "up" state.
+    3. Enslave interface to the created bridge entity.
+    4. Insert interface to all VLANs possible(4094)
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: space between "possible" and "4094"

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    5. Verify interface is in all possible(4094) VLANs
+    """
+
+    max_vlans = 4094
+    #  1.Initiate test params
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 1)
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: If you do not need traffic, there is no reason to set the number of links to 1.
Also you can use just tb_get_all_devices which doesn't require TG in test config at all.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    device = dent_devices[0].host_name
+    ports = tgen_dev.links_dict[device][1]
+    test_port = ports[0]
+
+    #  2.Create bridge entity and set state to "up" state
+    await IpLink.add(input_data=[{device: [{"device": "br0", "type": "bridge", "vlan_filtering": 1}]}])
+    out = await IpLink.set(input_data=[{device: [{"device": "br0", "operstate": "up"}]}])
+    assert out[0][device]["rc"] == 0, f" Verify bridge entity created and bridge entity set to 'up' state.\n {out}"
+
+    await IpLink.add(input_data=[{device: [{"device": f"{test_port}", "type": "veth"}]}])
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: Why are you adding veth pairs?

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    out = await IpLink.set(input_data=[{device: [{"device": f"{test_port}", "operstate": "up"}]}])
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: You can combine this command with the next one on line 44.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    # 5. Verify interface is in all possible(4094) VLANs
+    out = await BridgeVlan.show(input_data=[{device: [{"device": test_port, "cmd_options": "-j"}]}])
+    vlans = json.loads(out[0][device]["result"])
+    assert len(vlans[0]["vlans"]) == max_vlans, f"Not all VLANS has been added to {test_port} on {device} device \n"
+
+
+@pytest.mark.asyncio
+async def test_add_interface_to_vlan_wo_bridge(testbed):
+    """
+    Test Name: Add  interface to vlan without enslaving to bridge entity
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: Please remove extra space after "Add"
Also add some mark that this is negative test.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    1. Initiate test params
+    2. Insert interface to any VLAN
+       Verify adding interface to VLAN fails
+    3. Create bridge entity.
+    4. Insert interface to any VLAN
+       Verify adding interface to VLAN fails
+    """
+
+    # 1.Initiate test params
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 1)
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: See my comments above about using TG here.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    ports = tgen_dev.links_dict[device][1]
+    test_port = ports[0]
+    out = await IpLink.add(input_data=[{device: [{"device": f"{test_port}", "type": "veth"}]}])
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_without_traffic.py]: Please remove veth creation.

**Review by anton7811 — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    assert "RTNETLINK answers: Operation not supported" in out[0][device]["result"]
+
+    # 3. Create bridge entity.
+    await IpLink.add(input_data=[{device: [{"device": "br0", "type": "bridge", "vlan_filtering": 1}]}])
+    out = await IpLink.set(input_data=[{device: [{"device": "br0", "operstate": "up"}]}])
+    assert out[0][device]["rc"] == 0
+
+    # Insert interface to any VLAN. Verify adding interface to VLAN fails
+    out = await BridgeVlan.add(input_data=[{device: [{"device": test_port, "vid": 2}]}])
+    assert "RTNETLINK answers: Operation not supported" in out[0][device]["result"]
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py:83]: No new line at the end of the file. See PEP8

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    3. Enslave interface to the created bridge entity.
+    4. Insert interface to all VLANs possible (4094)
+    5. Verify interface is in all possible (4094) VLANs
+    """
+
+    max_vlans = 4094
+    #  1.Initiate test params
+    dent_devices = await tb_get_all_devices(testbed)
+    device = dent_devices[0].host_name
+    test_port = dent_devices.links_dict[device][1][0]  # get first port from config
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py:29]: In fact, you might not have any port in config. It should be enough just to use "swpX"
But let's leave as is so far.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+    #  2.Create bridge entity and set state to "up" state
+    await IpLink.add(input_data=[{device: [{"device": "br0", "type": "bridge", "vlan_filtering": 1}]}])
+    out = await IpLink.set(input_data=[{device: [{"device": "br0", "operstate": "up"}]}])
+    assert out[0][device]["rc"] == 0, f" Verify bridge entity created and bridge entity set to 'up' state.\n {out}"
+
+    # 3. Enslave interface to the created bridge entity
+    await IpLink.set(input_data=[{device: [{"device": f"{test_port}", "master": "br0"},
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py:37]: You can do this in a single command:
`{"device": f"{test_port}", "master": "br0", "operstate": "up"}`
But yet again. In this particular case you can leave it as is

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+                                           {"device": f"{test_port}", "operstate": "up"}
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_configuration_checks.py:38]: Please do not use constructions like this `f"{test_port}"`
Here you substitute string to string with string to string conversion. Do just `{"device": test_port}`

**Review by anton7811 — CHANGES_REQUESTED** [review]

**Comment by StepanVovkPLV** [comment]
closing in favor of https://github.com/PLVision/dent-testing/pull/15

**Pull request closed without merging**